### ISSUE TITLE
CORS Error - NetworkError when attempting to fetch resource 

### ISSUE THREAD

**Issue by mikeaadd** [heartexlabs/label-studio#659]
Describe the bug
When uploading task with audio data, I receive CORS error message and am unable to listen to audio.

To Reproduce
   This is similar to #481. 
    1. start up label-studio (version 1.0)
    2. upload json tasks with with UI. An example task...
```
  {'data': {
    'transcript': "text goes here",
    'audio': 'http://127.0.0.1:81/05734d6b-074f-45b1-8af7-c1b1502ca31e.wav'}
  }
```
    3. spin up http-server with cors with command 
```
http-server -p 81 --cors
```
    4. Open up task to see if audio works and see that it fails
    
<img width="444" alt="Screen Shot 2021-03-17 at 3 05 33 AM" src="https://user-images.githubusercontent.com/12502807/111442286-cebf6700-86cd-11eb-9a3b-58e4d7910a18.png">

    5.  Confirm that cors is set correctly
    
 ```
(base) # curl http://localhost:81/ -I
HTTP/1.1 200 OK
access-control-allow-origin: *
access-control-allow-headers: Origin, X-Requested-With, Content-Type, Accept, Range
server: ecstatic-3.3.2
Content-Type: text/html
etag: W/"222381908-57344-2021-03-17T06:51:09.115Z"
last-modified: Wed, 17 Mar 2021 06:51:09 GMT
cache-control: max-age=3600
Date: Wed, 17 Mar 2021 09:07:40 GMT
Connection: keep-alive
Keep-Alive: timeout=5
Content-Length: 0
```

    6. Confirm that file is actually available at url.
   
```
    (base) # wget http://127.0.0.1:81/05734d6b-074f-45b1-8af7-c1b1502ca31e.wav
    --2021-03-17 09:09:52--  http://127.0.0.1:81/05734d6b-074f-45b1-8af7-c1b1502ca31e.wav
    Connecting to 127.0.0.1:81... connected.
    HTTP request sent, awaiting response... 200 OK
    Length: 24779452 (24M) [audio/wav]
    Saving to: ‘05734d6b-074f-45b1-8af7-c1b1502ca31e.wav’
    
    100%[==================================================================================================================================>] 24,779,452  --.-K/s   in 0.1s
    
    2021-03-17 09:09:52 (203 MB/s) - ‘05734d6b-074f-45b1-8af7-c1b1502ca31e.wav’ saved [24779452/24779452]
```

**Environment (please complete the following information):**
 - OS: Amazon Linux 2
 - Label Studio Version 1.0
   
**Additional context**
Ive tried using a nginx server as suggested to no avail. I'm not sure where else to go from here. Is this the easiest way to upload multimedia tasks?

**Comment by makseq** [comment]
@mikeaadd Why don't you drag & drop files into LS on Import page?

**Comment by makseq** [comment]
@mikeaadd What if you use `http://localhost:81/05734d6b-074f-45b1-8af7-c1b1502ca31e.wav'` instead of `http://127.0.0.1:81/05734d6b-074f-45b1-8af7-c1b1502ca31e.wav` ?

**Comment by mikeaadd** [comment]
I have tried both locahost and 127.0.0.1. As far as why i don't drag and drop. Im not sure how that would work. My task requires both audio and text. Dragging and dropping audio and text separately doesn't seem to work?

**Comment by mikeaadd** [comment]
Update on this journey:
1. I changed the port to 8080 and it didn't but looking at the logs you could tell I was actually receiving requests. 
2. I changed my browser from firefox to chrome and it worked! I had a colleague test it and she got the previous cors error. We checked our browser versions. She had 89, I had 88. I upgraded and now it doesn't work again.

**Comment by makseq** [comment]
Thank you very much for your report. I think it's your server with images issue. Could you try to use nginx to serve your media files?

**Comment by mikeaadd** [comment]
So i have tried setting up an nginx server to serve the files on the same server that label studio is being stood up. The relevant nginx configuration is below

```
    server {
        listen       8080;
        autoindex on;

        location / {
            root /label_studio_data/mp3s;
            # try_files $uri $uri/ =404;

            if ($request_method = 'OPTIONS') {
                add_header 'Access-Control-Allow-Origin' '*';
                add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
                #
                # Custom headers and headers various browsers *should* be OK with but aren't
                #
                add_header 'Access-Control-Allow-Headers' 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range';
                #
                # Tell client that this pre-flight info is valid for 20 days
                #
                add_header 'Access-Control-Max-Age' 1728000;
                add_header 'Content-Type' 'text/plain; charset=utf-8';
                add_header 'Content-Length' 0;
                return 204;
            }
            if ($request_method = 'POST') {
                add_header 'Access-Control-Allow-Origin' '*';
                add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
                add_header 'Access-Control-Allow-Headers' 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range';
                add_header 'Access-Control-Expose-Headers' 'Content-Length,Content-Range';
            }
            if ($request_method = 'GET') {
                add_header 'Access-Control-Allow-Origin' '*';
                add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
                add_header 'Access-Control-Allow-Headers' 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range';
                add_header 'Access-Control-Expose-Headers' 'Content-Length,Content-Range';
            }
        }
    }
```

A sample json task looks like so...

```
{'transcript': "transcript goes here",
 'audio': 'http://localhost:8080/fd576c11-d9e3-445f-852e-68a3c55cd1b4.mp3'}
```

I still get the same CORS error

**Comment by makseq** [comment]
@mikeaadd What port do you use for label studio?

**Comment by mikeaadd** [comment]
So i finally have been able to get this working. I used port 80 for label studio and 8080 for the file server. When I used the public ip address it did not work initially. I stupidly forgot that it was behind a vpc so I didn't have the right ip address. Once that was resolved, it seems to work fine now. I'm still a little confused at why localhost worked even if it was intermittently, but I seem to have a stable system at the moment.
